Cython/Compiler/ExprNodes.py

```diff
@@ -9916,6 +9916,7 @@ def analyse_declarations(self, env):
         self.def_node.no_assignment_synthesis = True
         self.def_node.pymethdef_required = True
         self.def_node.is_cyfunction = True
+        self.code_object = CodeObjectNode(self.def_node)
```

Comment:
@da-woods, I've added this line but I don't understand the full working or if it's correct. I see this but I want to know how I can write tests and go further with this.

```py
    @classmethod
    def from_defnode(cls, node, binding):
        return cls(node.pos,
                   def_node=node,
                   pymethdef_cname=node.entry.pymethdef_cname,
                   binding=binding or node.specialized_cpdefs,
                   specialized_cpdefs=node.specialized_cpdefs,
                   code_object=CodeObjectNode(node))

```

Reply:
And it doesn't compile so this is definitely wrong 

`AttributeError: 'GeneratorDefNode' object has no attribute 'local_scope'`

Reply:
It's quite likely you just need to move your added line slightly later so that it's after `self.def_node.analyse_declarations(env)` (which i think in where `local_scope` is set up).

In terms of testing, start by cherry-picking the commit I suggested that @scoder has made containing some preliminary tests for this. Any tests you add will want to go in the same file. You can run the tests with `python runtests.py <filename_without_the_extension>`. In terms of adding tests, see if there's anything you missed or if there's any way you can think of breaking it.

It's possible that this change also adds a signature to generators (since Cython shares a lot of code between generators and lambda functions) so maybe look at making sure that works.